internal/storage/persistence/ent/client/client.go

```diff
@@ -65,15 +68,59 @@ func (c *Client) CreateAIP(ctx context.Context, goaaip *goastorage.AIP) (*goasto
 	return aipAsGoa(ctx, a), nil
 }
 
-func (c *Client) ListAIPs(ctx context.Context) (goastorage.AIPCollection, error) {
-	aips := []*goastorage.AIP{}
+func (c *Client) ListAIPs(ctx context.Context, payload *goastorage.ListAipsPayload) (*goastorage.AIPs, error) {
+	if payload == nil {
+		payload = &goastorage.ListAipsPayload{}
+	}
+
+	createdAt, err := timerange.Parse(payload.EarliestCreatedTime, payload.LatestCreatedTime)
+	if err != nil {
+		return nil, goastorage.MakeNotValid(fmt.Errorf("created at: %v", err))
+	}
+
+	var status *enums.AIPStatus
+	if payload.Status != nil {
+		s, err := enums.ParseAIPStatus(*payload.Status)
+		if err != nil {
+			return nil, goastorage.MakeNotValid(errors.New("status: invalid value"))
+		}
+		status = &s
+	}
 
-	res, err := c.c.AIP.Query().All(ctx)
+	qf := entfilter.NewFilter(c.c.AIP.Query(), entfilter.SortableFields{
+		aip.FieldID: {Name: "ID", Default: true},
+	})
+	qf.Contains(aip.FieldName, payload.Name)
+	qf.Equals(aip.FieldStatus, status)
+	qf.AddDateRange(aip.FieldCreatedAt, createdAt)
+	qf.OrderBy(entfilter.NewSort().AddCol("id", true))
+	qf.Page(ref.DerefZero(payload.Limit), ref.DerefZero(payload.Offset))
+	page, whole := qf.Apply()
+
+	res, err := page.All(ctx)
+	if err != nil {
+		return nil, goastorage.MakeNotAvailable(errors.New("cannot perform operation"))
```

Comment:
Also, it looks like we are returning `409` instead of `500` errors across the entire API, not sure if there was a reason for that, but we may want to consider changing that at some point. For now, I kept the same behavior in this endpoint.

Reply:
My impression from working with the Goa errors for issue #1141 is that the error handling code was "roughed in" but wasn't tested to make sure the HTTP response codes were consistent with the error that produced them.  I think part of the problem is that Goa error handling is not well documented, and there is mixing of different layers of the application --- like in this case where an API error (`goastorage.MakeNotAvailable()`) is being created in persistence layer.

Reply:
But there is a `StatusInternalServerError` error type available, that I think fits better in these cases.

Reply:
I agree that a 500 HTTP error response is better here, I just want to make sure we aren't returning 500 error codes in cases where it's not appropriate (e.g. when a invalid request parameter value is sent).